README.adoc

```diff
@@ -0,0 +1,168 @@
+
+image::docs/images/ChatUI_NavigationGuide.png[Chat Application - Navigation Guide, align="center"]
+{nbsp} +
+
+== Code implementation
+
+[horizontal]
+chat-ui:: directory containing the code of the stateless service `chat`. This code is written in TypeScript, and run as a Node.js server.
+deploy:: directory containing the yaml files to deploy all the services composing the Chat application. The target cluster could be your own Kubernetes cluster or https://docs.lbcs.dev/index.html[Lightbend Cloudstate]
+
+The code of the backend services `Friends` and `Presence` is separate from the UI code. This is intentional to demonstrate the polyglot aspect of Cloudstate services. They can seamlessly interact regardless of the programming language used to implement the service. Below are the available implementations. You can choose any combination for deployment. For example, `Friends` service in JavaScript and `Presence` service in Java.
+
+=== JavaScript implementation
+
+* https://github.com/cloudstateio/samples-js-chat/tree/master/friends[Friends service]
+* https://github.com/cloudstateio/samples-js-chat/tree/master/presence[Presence service]
+
+=== Java implementation
+
+* https://github.com/cloudstateio/samples-java-chat/tree/master/friends[Friends service]
+* https://github.com/cloudstateio/samples-java-chat/tree/master/presence[Presence service]
+
+
+== Prerequisites
+
+=== JavaScript Dev environment
+
+https://github.com/cloudstateio/samples-js-chat/blob/master/README.adoc#js-devenv-setup-for-cloudstate[JavaScript Dev environment setup for Cloudstate]
+
+[[prereq-docker-registry]]
+=== Docker repository
+
+You must package your services into docker images and publish these images in a Docker registry. At deployment time, the Kubernetes cluster will pull the images from this registry. You can get a free Docker registry at https://hub.docker.com/
+
+
+=== Self-hosting or Managed hosting?
+
+For the deployment, there are two possible choices. Below we highlight the main characteristics of each deployment approach. Details about prerequisites and infrastructure preparation would be discussed further in the <<deployment,Deployment>> section. 
+
+==== Managed hosting
+
+This deployment approach doesn't require any local infrastructure. Lightbend Cloudstate will take care of the hosting, configuration and maintenance of your services. The deployment prodedure is pretty minimal:
+
+* Open an account with Lightbend Cloudstate
+* Install the "Cloudstate CLI" tool `csctl`
+
+And that's it. Then you just use `csctl` to deploy the your services that you had published in your Docker registry.
+
+ 
+==== Self-hosting
+
+In case you prefer to host your own services. You need to provide a Kubernetes cluster. This cluster could be local or cloud-based and must fulfill the following requirements:
```

Comment:
```suggestion
In case you prefer to host your own services, you need to provide a Kubernetes cluster. This cluster could be local or cloud-based and must fulfill the following requirements:
```